chore: add preset option in create new form and remove from preferences

SoniaSandler opened the pull request:
This PR removes the option to set the preset in preferences, and adds a preset field in the create new form.
Using the audit method, the preset is updated on each preset change in the form before submission. With each preset change, some of the other form fields also change to match the correct preset. 
The preset change is persistent as it triggers each time a config change.


https://github.com/user-attachments/assets/c831511e-4bf0-4c61-9d3c-b1d5916c8601



Closes https://github.com/crc-org/crc-extension/issues/905

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Configuration Changes**
  * Added crc.factory.preset, removed legacy OpenShift-Local.preset; consolidated scopes and added defaults for CRC settings.

* **New Features**
  * Connection auditor detects and applies CRC presets via CLI and emits preset change events.

* **Improvements**
  * CRC setup streamlined to a non-interactive flow; preset persisted and user messages shortened.

* **Tests**
  * Removed Preferences UI tests; added tests validating auditor-driven preset application.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

gastoner reviewed (commented):
On Linux when I'm on dashboard it no longer offers me one of two options openshift/microshift this selection is done by `crc config` from console which I feel like step back ? also not sure if this was in the scope of this issue

<img width="1718" height="776" alt="Image" src="https://github.com/user-attachments/assets/9ed2c76a-75aa-4c3d-a629-e56b550b768a" />

Also I'm able to get to the preference from resources page, although the option is not visible in preferences, changing this value will update the settings in `settings.json` file

https://github.com/user-attachments/assets/11f965ff-af11-4a59-a485-c1bcc8e13470


I think that the functionality on dashboard should stay the same (when creating select one or another) + just add mechanism to set the preset in our config (in global one is probably another issue) 

Also thinking that we should not see the settings in resources 
WDYT @jeffmaury @odockal

SoniaSandler commented:
> On Linux when I'm on dashboard it no longer offers me one of two options openshift/microshift this selection is done by crc config from console which I feel like step back ? also not sure if this was in the scope of this issue

@gastoner I'm not sure I understand what you mean about this

> Also I'm able to get to the preference from resources page, although the option is not visible in preferences, changing this value will update the settings in settings.json file

Thanks for catching this! I have updated the PR

SoniaSandler commented on src/preferences.ts:
```diff
@@ -140,141 +84,37 @@ async function handleProxyChange(proxy?: extensionApi.ProxySettings): Promise<vo
-      }
-    }
+  const preset = initialCrcConfig.preset ?? 'openshift';
+  if (preset !== 'podman') {
```
So I checked again, and `refreshConfig` is needed to update the starting values in the form.
When submitting the form, we're using the submitted values to update the crc config, and then from there we use the updated crc config to update the starting values of the form. However, I have removed `await extConfig.update('crc.factory.preset', initialCrcConfig.preset);` that I added in this PR since we do this update in another place too
With `refreshConfig`: 

https://github.com/user-attachments/assets/1ee99f4e-c648-4eca-9567-24f1d5b20275

Without `refreshConfig`:

https://github.com/user-attachments/assets/1e0e40f7-768f-4b6e-b3f4-7cd11bf1f5a3

bmahabirbu reviewed (commented):
I confirmed @gastoner for linux (specifically im on fedora 42) that i no longer see the dialog option for selecting microshift vs openshift after the initialization on the dashboard

bmahabirbu commented:
<img width="1735" height="245" alt="Screenshot From 2025-12-04 21-13-58" src="https://github.com/user-attachments/assets/9099bd0c-9de1-4178-9f1b-b330d233c094" />

Not related to this pr (i think) but if I get the error cluster is busy the inlitzation drop down disappears even when going from a page back to the dashboard

bmahabirbu commented:
confirmed selection works as intended!

SoniaSandler commented on src/crc-setup.ts:
```diff
@@ -30,28 +30,7 @@ export async function needSetup(): Promise<boolean> {
-      await execPromise(getCrcCli(), ['config', 'set', 'preset', choice]);
-    }
-  }
-
```
@gastoner I re-added the code, so it should be back to normal now and show the option to choose openshift or microshift

jeffmaury commented:
> Tried again now, but got into quite an odd thing
> 
> Steps to reproduce: run `crc stop` `crc delete` and `crc cleanup` => then you should see on the dashboard dropdown, chose just `Init` (without start) and pick `Microshift` This should fail:
> <img alt="Image" width="1214" height="811" src="https://private-user-images.githubusercontent.com/46690189/522909770-dd5b07f8-9aa6-437d-8d2a-3a636dec2fc7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ5Mzg2NDQsIm5iZiI6MTc2NDkzODM0NCwicGF0aCI6Ii80NjY5MDE4OS81MjI5MDk3NzAtZGQ1YjA3ZjgtOWFhNi00MzdkLThkMmEtM2E2MzZkZWMyZmM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjA1VDEyMzkwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYjdlNjQ0MGRjMWZjNzcyOTg2Y2I4OWIwMmJhMmM1YWJhMjhjM2YwOGMzNTg2ZGU4Y2ViYjk2MDRkODdhZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.praOfRKwcRo-pli7nfntrjWI4flAIBsGrhyoggZrbyk">
> 
> Go to resources and try to create the VM => it will offer you the pick one or another dialog which should not be there since you have the dropdown for it already:
> <img alt="Image" width="1214" height="811" src="https://private-user-images.githubusercontent.com/46690189/522908128-1f30dd31-55a0-4e6e-8806-543733e3a169.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ5Mzg2NDQsIm5iZiI6MTc2NDkzODM0NCwicGF0aCI6Ii80NjY5MDE4OS81MjI5MDgxMjgtMWYzMGRkMzEtNTVhMC00ZTZlLTg4MDYtNTQzNzMzZTNhMTY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjA1VDEyMzkwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1NDEzOWNkMjJhODcyMmJkZmFhYWRlNzI5NjcyN2IxNzE0NDhkYjljMDNlNGU0YWJmY2ZhMmQ1YjUzOTM4MTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PPod0FLyb7i08jbaisO4QKVCM221sAuNgCPfY6kvNTI">
> 
> Tried this on main and it has the same issue, but since we have the preset dropdown now, we should not see this message box. So, thinking if we want to tackle this in this PR too cc @jeffmaury @SoniaSandler WDYT?
> 
> Otherwise LGTM on Linux

Yes the preset selection dialog should be removed

SoniaSandler commented:
@jeffmaury @gastoner I have updated the PR, removed the preset dialog, and now the form values (except for the preset value which should show at the beginning as the value from the crc config) are not saved for new forms
